### PR TITLE
Use `python3` as the Python path within containers

### LINKED ISSUE #2455: Debug Python within a container  Couldn't spawn debuggee
I'm following the instructions as for https://code.visualstudio.com/docs/containers/debug-python

this used to work before upgrading docker to version 19.03.13

Python 3.7.6
Docker version 19.03.13, build 4484c46d9d

Create a project folder
`mkdir my-project && cd my-project`

Create a minimal python file named app.py
`
def main():
    print('HELLO')

if __name__ == "__main__":
    main()
`

Create a python3 virtual environment.
`python3 -m venv py-venv`

setting.json
`
{
    "python.pythonPath": "py-venv/bin/python"
}
`

tasks.json
`{
	"version": "2.0.0",
	"tasks": [
		{
			"type": "docker-build",
			"label": "docker-build",
			"platform": "python",
			"dockerBuild": {
				"tag": "pydockerdebug:latest",
				"dockerfile": "${workspaceFolder}/Dockerfile",
				"context": "${workspaceFolder}",
				"pull": true
			}
		},
		{
			"type": "docker-run",
			"label": "docker-run: debug",
			"dependsOn": [
				"docker-build"
			],
			"python": {
				"file": "app.py"
			}
		}
	]
}
`

launch.json
`
{
    "configurations": [
        {
            "name": "Docker: Python - General",
            "type": "docker",
            "request": "launch",
            "preLaunchTask": "docker-run: debug",
            "python": {
                "pathMappings": [
                    {
                        "localRoot": "${workspaceFolder}",
                        "remoteRoot": "/app"
                    }
                ],
                "projectType": "general"
            }
        }
    ]
}
`

Create a Dockerfile
`
FROM python:3.8-slim-buster
ENV PYTHONDONTWRITEBYTECODE=1
ENV PYTHONUNBUFFERED=1

WORKDIR /app
ADD . /app

RUN useradd appuser && chown -R appuser /app
USER appuser

CMD ["python", "app.py"]
`

Run Debug 

and the following error occurs

Couldn't spawn debuggee: [Errno 2] No such file or directory: '/Users/.../py_docker_debug/py-venv/bin/python'

Command line:['/Users/.../py_docker_debug/py-venv/bin/python', '/debugpy', '--connect', 'host.docker.internal:52703', '--configure-qt', 'auto', '--adapter-access-token', '875b8dec0efaabdcdfad7028575706dbb916c2d8bde6e3fd8c25da2142db2a8a', 'app.py']

### PR DESCRIPTION
**Pull request by bwateratmsft**
According to [this comment](https://github.com/microsoft/vscode-docker/issues/2455#issuecomment-724267419), this should be the changes needed on the Docker side to fix #2455.

@int19h, if I understand the flow correctly, I think that the Python extension handles `debugAdapterPython` and `debugLauncherPython`. The `debugLauncherPython` Python interpreter will start our "launcher", which then unconditionally uses `python3` to start the real debugpy launcher inside the container: https://github.com/microsoft/vscode-docker/blob/master/resources/python/launcher.py#L17

Then, using the `"python": "python3"` option in this PR, the launcher inside the container uses `python3` to launch the app, which is what is desired.

Does all this sound right?

### CODE REVIEW

File: src/tasks/python/PythonExtensionHelper.ts [hunk 1]
```diff
@@ -36,7 +36,7 @@ export namespace PythonExtensionHelper {
 
     export async function getPythonExtension(): Promise<vscode.Extension<PythonExtensionAPI>> | undefined {
         const pyExtensionId = 'ms-python.python';
-        const minPyExtensionVersion = new semver.SemVer('2020.5.78807');
+        const minPyExtensionVersion = new semver.SemVer('2020.5.78807'); // TODO
```
Comment (philliphoff) [inline src/tasks/python/PythonExtensionHelper.ts:39]: Maybe add more of a comment as to what's needed to do?

**Review by philliphoff — APPROVED** [review]

File: src/tasks/python/PythonExtensionHelper.ts (hunk 1 above)
Comment (bwateratmsft) [reply src/tasks/python/PythonExtensionHelper.ts:39]: Yeah, I put this in to remember to bump the new min version before submitting. I'm still trying to figure out what the new min version ought to be.

**Comment by int19h** [comment]
Yep, once we merge https://github.com/microsoft/vscode-python/pull/14676 and https://github.com/microsoft/debugpy/pull/467/, this change should do the trick for you.

**Comment by int19h** [comment]
Here's the extension build with the requisite change:
https://github.com/microsoft/vscode-python/releases/tag/2020.11.367453362

**Pull request merged**